### DISCUSSION TITLE
能支持添加自定义 question 吗？

### DISCUSSION THREAD

**Discussion by AnsonKong** [Zhengqbbb/cz-git#172] [General]
目前有个场景是自定义 question 需要基于 scope 的值去判断是否需要显示，但是目前看配置并没有相关的配置。

**Comment by Zhengqbbb** [comment]
不确定自定义 question 是指什么：
1. 是 question 显示内容修改：https://cz-git.qbb.sh/zh/config/show#messages
2. 是否显示 question：https://cz-git.qbb.sh/zh/config/engineer#skipquestions

---

基于 scope，也就是说是用户 scope 使用了什么，从而动态判断是否显示后续的 question 吗？
1. 目前只有基于 type，动态显示 scope 选择列表：https://cz-git.qbb.sh/zh/config/engineer#scopeoverrides
2. 基于 scope 动态显示或不显示什么 question，看看是否很多场景需求，还是比较个性化的习惯 (我可能不会额外增加配置的复杂性)
3. **如果是比较个性化的需求，可以尝试看看是否可以根据其他 js 方式获取对于 question 显示的判断条件，从而动态改变 `skipQuestions` 这个配置数组，来达到需求**：
     - https://cz-git.qbb.sh/zh/config/engineer#skipquestions
     - 提前获取信息，从而动态改变配置玩法大致可以参照：
         - [https://cz-git.qbb.sh/zh/recipes](https://cz-git.qbb.sh/zh/recipes/#%E9%92%88%E5%AF%B9-%E9%A1%B9%E7%9B%AE%E4%BB%A3%E7%A0%81-%E7%9A%84-scopes)
         - https://cz-git.qbb.sh/zh/recipes/default-issues

希望可以帮助到你

**Reply by AnsonKong** [reply]
可能你没理解我的问题，我看了下你的代码 `packages/cz-git/src/generator/question.ts` 里面的 `generateQuestions` 函数是写死了返回的 `questions` 数组的，而我需要在中间插入一个 `question`。`skipQuestions` 只能做减法，做不了加法。

**Reply by Zhengqbbb** [reply]
抱歉不支持

1. 可 fork 代码自行处理添加
2. 可选择一个不常用的 question 进行改造：https://cz-git.qbb.sh/zh/config/engineer#formatmessagecb

**Reply by AnsonKong** [reply]
我需要在 `customScope` 下方添加 question，估计第二个方案不行
